frontend/src/pages/private/products/components/DataTableProducts.jsx

```diff
@@ -0,0 +1,171 @@
+import { DeleteRounded, EditRounded, Visibility } from '@mui/icons-material';
+import { Box, IconButton } from '@mui/material';
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect, useState } from 'react';
+import CustomDialog from '@/components/customDialog/CustomDialog.component';
+import DataTable from '@/components/dataTable/DataTable';
+import ModalProductDetails from './ModalProductDetails';
+import stock from '@/redux/stock/actions';
+import Loading from '@/components/Loading';
+
+const DataTableProducts = () => {
+  const dispatch = useDispatch();
+  const [openModal, setOpenModal] = useState(false);
+  const [isUpdate, setIsUpdate] = useState(false);
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [selectedRow, setSelectedRow] = useState({
+    id: '',
+    name: '',
+  });
+
+  const handleDetails = async (id) => {
+    setSelectedRow({ ...selectedRow, id });
+    await dispatch(stock.read({ entity: 'stock', id }));
+    setIsUpdate(false);
+    setOpenModal(true);
+  };
+
+  const handleEdit = async (id) => {
+    setSelectedRow({ ...selectedRow, id });
+    await dispatch(stock.read({ entity: 'stock', id }));
+    setIsUpdate(true);
+    setOpenModal(true);
+  };
+
+  const handleDisable = (id, name) => {
+    setSelectedRow({ ...selectedRow, id, name });
+    setDialogOpen(true);
+  };
+
+  const handleDialogCancel = () => {
+    setDialogOpen(false);
+  };
+
+  const handleDialogAccept = async () => {
+    await dispatch(stock.delete({ entity: 'stock', id: selectedRow.id }));
+    setDialogOpen(false);
+  };
+
+  const productState = useSelector((store) => store.stock.listAll);
+  const readProductState = useSelector((store) => store.stock.read);
+  const deleteProductState = useSelector((store) => store.stock.delete);
+  const updateProductState = useSelector((store) => store.stock.update);
+
+  const [rows, setRows] = useState([]);
+
+  useEffect(() => {
+    if (productState?.result) {
+      setRows(productState.result.items.result);
+    }
+  }, [productState]);
+
+  const updateTable = async () => {
+    if (productState?.isLoading) return;
+    dispatch(stock.listAll({ entity: 'stock' }));
+  };
+
+  useEffect(() => {
+    updateTable();
+  }, [deleteProductState, updateProductState]);
+
+  const columns = [
+    {
+      field: 'imageUrl',
+      headerName: 'Foto',
+      width: 100,
+      renderCell: (params) => {
+        const { id } = params.row;
+        return (
+          <Box display="flex" width="100%" onClick={() => handleDetails(id)} alignItems="center">
+            <img
+              src={params.value ? params?.value : '/noImage.png'}
```

Comment:
en el caso que value no exista este codigo se va a romper
cannot read property value of undefined
El ? del segundo termino no es necesario porque si entra ahi ya se sabe que existe
seria:

params?.value ? params.value : '/noImage.png'

o mas correcto
params?.value ?? '/noimage.png'

Reply:
params?.value ?? '/noimage.png'
de esta forma no renderiza el noimage.png